pkg/syncer/mutators/deployment.go

```diff
@@ -0,0 +1,247 @@
+package mutators
+
+import (
+	"net"
+	"net/url"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/rest"
+	utilspointer "k8s.io/utils/pointer"
+)
+
+type DeploymentMutator struct {
+	fromConfig *rest.Config
+	kcpHost    string
+}
+
+func (dt *DeploymentMutator) getGVR() schema.GroupVersionResource {
+	return schema.GroupVersionResource{
+		Group:    "apps",
+		Version:  "v1",
+		Resource: "deployments",
+	}
+}
+
+func (dt *DeploymentMutator) Register(mutators map[schema.GroupVersionResource]Mutator) {
+	if _, ok := mutators[dt.getGVR()]; !ok {
+		mutators[dt.getGVR()] = dt
+	}
+}
```

Comment:
this is an odd pattern. Why should a mutator know how to register into some map of mutators?